ui/prompt.go

```diff
@@ -44,18 +48,141 @@ func PromptProjects(projects []*entity.Project) (*entity.Project, error) {
 		Templates: &promptui.SelectTemplates{
 			Active:   `{{ .Name | underline }}`,
 			Inactive: `{{ .Name }}`,
-			Selected: fmt.Sprintf("%s Project: {{ .Name | magenta | bold }} ", GreenText("✔")),
+			Selected: fmt.Sprintf("%s Project: {{ .Name | magenta | bold }} ", promptui.IconGood),
 		},
 	}
 	i, _, err := prompt.Run()
 	return projects[i], err
 }
 
+// PromptStarterTemplates fetches available templates and prompts the user to select one
+func PromptStarterTemplates() (*entity.Template, error) {
+	StartSpinner(&SpinnerCfg{
+		Message: "Fetching starter templates",
+	})
+	resp, err := http.Get("https://raw.githubusercontent.com/railwayapp/starters/master/featured.json")
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var data struct {
+		Templates []entity.Template `json:"examples"`
+	}
+	dec := json.NewDecoder(resp.Body)
+	err = dec.Decode(&data)
+	if err != nil {
+		return nil, err
+	}
+
+	StopSpinner("")
+
+	prompt := promptui.Select{
+		Label: "Select Starter Template",
+		Items: data.Templates,
+		Templates: &promptui.SelectTemplates{
+			Active:   fmt.Sprintf("%s {{ .Text | underline }}", promptui.IconSelect),
+			Inactive: `  {{ .Text }}`,
+			Selected: fmt.Sprintf("%s Template: {{ .Text | magenta | bold }} ", GreenText("✔")),
+		},
+	}
+	i, _, err := prompt.Run()
+	return &data.Templates[i], err
+}
+
+func PromptIsRepoPrivate() (bool, error) {
+	prompt := promptui.Select{
+		Label: "Select repo visibility",
+		Items: []string{"Public", "Private"},
+	}
+	_, visibility, err := prompt.Run()
+	return visibility == "Private", err
+}
+
+func PromptEnvVars(envVars []entity.TemplateEnvVar) (map[string]string, error) {
+	variables := make(map[string]string)
+	if len(envVars) > 0 {
+		fmt.Printf("\n%s\n", Bold("Environment Variables"))
+	}
+
+	requiredValidator := func(s string) error {
+		if s == "" {
```

Comment:
suggestion: trim the value here. A bunch of whitespace should be invalid for a required env var